lwb/metalang/sdf3/sdf3/src/main/java/mb/sdf3/stratego/Sdf3SslExtPlaceholderCharsPrimitive.java

```diff
@@ -0,0 +1,33 @@
+package mb.sdf3.stratego;
+
+import org.spoofax.interpreter.library.AbstractPrimitive;
+import mb.stratego.common.AdaptableContext;
+import org.spoofax.interpreter.core.IContext;
+import org.spoofax.interpreter.core.InterpreterException;
+import org.spoofax.interpreter.stratego.Strategy;
+import org.spoofax.interpreter.terms.IStrategoTerm;
+import org.spoofax.interpreter.terms.ITermFactory;
+
+/**
+ * Returns a tuple with the prefix and suffix, respectively, to use to parse/pretty-print placeholders.
+ */
+public final class Sdf3SslExtPlaceholderCharsPrimitive extends AbstractPrimitive {
+
+    public Sdf3SslExtPlaceholderCharsPrimitive() {
+        super("SSL_EXT_placeholder_chars", 0, 0);
+    }
+
+    @Override public boolean call(IContext env, Strategy[] svars, IStrategoTerm[] tvars) throws InterpreterException {
+        final ITermFactory factory = env.getFactory();
+        try {
+            final Sdf3Context context = AdaptableContext.adaptContextObject(env.contextObject(), Sdf3Context.class);
+            env.setCurrent(factory.makeTuple(
+                factory.makeString(context.placeholderPrefix),
+                factory.makeString(context.placeholderSuffix)
+            ));
+            return true;
+        } catch(RuntimeException e) {
+            return false; // Context not available; fail
```

Comment:
Catching all `RuntimeException`s is bad practice, as it catches all kinds of exceptions indicating bugs such as `NullPointerException`. This will silently hide them, which is very bad.

Reply:
This was based on your own code for [`Sdf3PpLanguageSpecNamePrimitive`](https://github.com/metaborg/spoofax-pie/blob/4072554a4adc376db351e0d3e6719db4c940e260/lwb/metalang/sdf3/sdf3/src/main/java/mb/sdf3/stratego/Sdf3PpLanguageSpecNamePrimitive.java).

Reply:
Then that code should be changed as well. I think it should catch `AdaptException` instead.